cmake/DD4hepBuild.cmake

```diff
@@ -36,7 +36,7 @@ endmacro(dd4hep_to_parent_scope)
 macro(dd4hep_set_compiler_flags)
   include(CheckCXXCompilerFlag)
 
-  SET(COMPILER_FLAGS -Wall -Wextra -pedantic -Wshadow -Wformat-security -Wno-long-long -Wdeprecated -fdiagnostics-color=auto)
+  SET(COMPILER_FLAGS -Wshadow -Wformat-security -Wno-long-long -Wdeprecated -fdiagnostics-color=auto -Wall -Wextra -pedantic)
```

Comment:
What does this re-shuffling change?

Reply:
Apparently nothing. I had the suspicion that -Wall or -Wextra after -Wshadow disables -Wshadow.
Note that the real order when compiling is inverted.
Net result is that -Wshadow does not work at all for gcc. Only for clang.